app/models/user.rb

```diff
@@ -12,7 +12,15 @@ class User < ActiveRecord::Base
 
   validates :name, :gender, presence: true
 
+  before_create :create_onesignal
```

Comment:
In my opinion a `before_create` callback isn't the best option, imagine that `OneSignal` service is broken or takes a lot of time. Your server will be blocked during the user sign up, which is terrible.

Since `id_onesignal` isn't key attribute during the sign up I would move this to a `after_create` callback and will run `NotificationService` and update the attribute in a job.

Be careful when you use external or third part services during request. Usually we don't want to block the server unnecessarily.